backend/games/wordGuesser/ClassicWordle.py

```diff
@@ -0,0 +1,33 @@
+from typing import List
+
+
+class ClassicWordle:
+    # maximum number of guesses
+    MAX_GUESSES = 6
+    # represents the state of the game, index 0 will hold the "goal word"
+    gameBoard = List[str]
+
+    # clears any previous activity and randomizes a word to initialize the list
+    def startGame(self) -> List[str]:
+        pass
+
+    # clears all previous progress and calls checkWin() to determine final results
+    # will return a victory or failure string
+    def endGame(self) -> str:
+        pass
+
+    # performs all logic checks on a word to determine if it is valid
+    # takes a word and updates the board based on
+    def guessWordle(self, guess: str) -> List[str]:
+        pass
+
+    # returns list of letters correctly guessed
+    # capital= right letter, right space
+    # lower = right letter wrong space
+    def trackProgress(self) -> List[str]:
+        pass
+
+    # checks to see if win/lose conditions have been satisfied
+    # meant to run after each guess, and after MAX_GUESS has been reached
+    def checkWin(self) -> bool:
+        pass
```

Comment:
Thanks for writing this!

Base on this, this is how I am planning to write the UI (frontend) and API (backend) flow that calls your game class:

1. User starts a game
  - UI starts listening to the database for their game data, although at first there is nothing
  -  UI sends a request to start a game
  - API calls your class to start a game
  - API assigns the game class instance to an ID for the chat
  - API saves an initial game state to the database
  - UI listens to this database update and displays the game board
2. User submits a guess
  - UI sends a request with the guess
  - API calls your class to submit the guess
  - API calls your class to check the win condition
  - API calls your class to track the progress
  - API saves the game state data to the database
  - UI listens to this database update and displays the game state
 3. The above continues until a game end condition is reached

Your code will fit into this flow. But one concern is that if the API needs to be restarted, we will lose all game class objects and the user will not be able to continue playing their game. In the future, you could consider designing a game class that is "stateless" or functional, in that it takes in any state data given to it, and can resume the game from there. But for this last week of the internship, your current approach should work.